tests/test_rules.py

```diff
@@ -0,0 +1,273 @@
+        with pytest.raises(TypeError):
+            ApplyRule(
+                "example.com",
+                *[params[r] for r in remove],
+                **{k: v for k, v in params.items() if k not in remove}  # type: ignore[arg-type]
+            )
+
+
+def test_list_page_objects_all() -> None:
+    rules = default_registry.get_rules()
+    page_objects = {po.use for po in rules}
+
+    # Note that the 'tests_extra.po_lib_sub_not_imported.POLibSubNotImported'
+    # Page Object is not included here since it was never imported anywhere in
+    # our test package. It would only be included if we run any of the following
+    # below. (Note that they should run before `get_rules` is called.)
+    #   - from tests_extra import po_lib_sub_not_imported
+    #   - import tests_extra.po_lib_sub_not_imported
+    #   - web_poet.consume_modules("tests_extra")
+    # Merely having `import tests_extra` won't work since the subpackages and
+    # modules needs to be traversed and imported as well.
+    assert all(["po_lib_sub_not_imported" not in po.__module__ for po in page_objects])
+
+    # Ensure that ALL Override Rules are returned as long as the given
+    # registry's @handle_urls decorator was used.
+    assert page_objects == POS.union({POLibSub})
+    for rule in rules:
+        # We're ignoring the types below since mypy expects ``Type[ItemPage]``
+        # which doesn't contain the ``expected_*`` fields in our tests.
+        assert rule.instead_of == rule.use.expected_instead_of, rule.use  # type: ignore[attr-defined]
+        assert rule.for_patterns == rule.use.expected_patterns, rule.use  # type: ignore[attr-defined]
+        assert rule.to_return == rule.use.expected_to_return, rule.use  # type: ignore[attr-defined]
+        assert rule.meta == rule.use.expected_meta, rule.use  # type: ignore[attr-defined]
+
+
+def test_registry_get_overrides_deprecation() -> None:
+    msg = "The 'get_overrides' method is deprecated. Use 'get_rules' instead."
+    with pytest.warns(DeprecationWarning, match=msg):
+        rules = default_registry.get_overrides()
+
+    # It should still work as usual
+    assert len(rules) == 15
+
+
+def test_consume_module_not_existing() -> None:
+    with pytest.raises(ImportError):
+        consume_modules("this_does_not_exist")
+
+
+def test_list_page_objects_all_consume() -> None:
+    """A test similar to the one above but calls ``consume_modules()`` to properly
+    load the ``@handle_urls`` decorators from other modules/packages.
+    """
+    consume_modules("tests_extra")
+    rules = default_registry.get_rules()
+    page_objects = {po.use for po in rules}
+    assert any(["po_lib_sub_not_imported" in po.__module__ for po in page_objects])
+
+
+def test_registry_search_rules() -> None:
+    rules = default_registry.search_rules(use=POTopLevel2)
+    assert len(rules) == 1
+    assert rules[0].use == POTopLevel2
+
+    rules = default_registry.search_rules(instead_of=POTopLevelOverriden2)
```

Comment:
Could you please add a test where search_rules is used 

1) with to_return argument?
2) with more than 1 search argument?

Reply:
Good idea! Updated in https://github.com/scrapinghub/web-poet/pull/84/commits/88c511d51cfdf64f1b4ae69563f7a97f8cf61cff.